### ISSUE TITLE
Editor not accessible with screen readers

### ISSUE THREAD

**Issue by RafaelFernandesBR** [laurent22/joplin#11328]
### Operating system

Windows

### Joplin version

3.1.22

### Desktop version info

Joplin 3.1.22 (prod, win32)

ID do cliente: ecdb6b3128d549d29e46c9f0743a3b09
Versão Sync: 3
Versão do Perfil: 47
Keychain Suportada: Sim

Revisão: da10297

Backup: 1.4.2
Describe image: 1.7.0

### Current behaviour

1. Select a note from the note list.

2. Open the note editor to make some edits to the text.

3. Check that, with a screen reader, the editor does not announce the lines of text when navigating with the up/down arrow. It also does not announce letters navigating right/left and words navigating with CTRL+right/left.

### Expected behaviour

Make it possible to read and navigate the text normally in the editor.

Another suggestion is to disable the option of going to the note title editor by pressing the up arrow on the first line of the editor.
This makes it confusing for those of us who use screen readers, and we end up editing the title thinking we are editing the body of the note.
If you want to maintain this functionality, add a warning to the screen reader, saying which editor we are in, when entering it.

**Comment by ayushpatil2122** [comment]
hey i want to work on this issue

**Comment by personalizedrefrigerator** [comment]
Thank you for reporting this! I'm summarizing my setup and what I've found so far.

## Attempting to reproduce this: Markdown editor

I haven't been able to reproduce this on Windows 11 with a custom build of Joplin [3.1.18](https://github.com/laurent22/joplin/releases/tag/v3.1.18). I've tested with two screen readers: NVDA and Windows Narrator. The below tests are with the **Markdown editor**.

Note that I plan to test with the release version of Joplin 3.1.23 in the near future &mdash; it's possible that this behavior is a recent regression. **Update**: I've just tested with 3.1.23 and the screen reader still behaves similar to the behavior detailed in the tests below.

For additional comparison, I'm including details of my tests with the two different screen readers:
<details><summary>NVDA test results</summary>

**Setup**: Joplin with a slightly modified version of the "1. Welcome to Joplin" note open in the editor. My cursor is at the beginning of the first paragraph. NVDA is enabled.

**Steps**:
1. Press the up arrow.
   - This moves the cursor to an empty line.
   - NVDA reads: "blank"
2. Press the up arrow again.
   - This moves the cursor to the Markdown heading, `# Welcome to Joplin!!`.
   - NVDA reads: "number Welcome to Joplin"
3. Press the right arrow.
   - This moves the cursor to just after the `#` character at the beginning of the Markdown heading.
   - NVDA reads: "space"
4. Press the right arrow.
   - This moves the cursor to just before the "W" in "Welcome to Joplin!!"
   - NVDA reads: "W" and plays a tone.
5. Press control, then the right arrow.
   - This moves the cursor to just after "Welcome".
   - NVDA reads: "Welcome".
6. Press the up arrow.
   - This moves the cursor to the start of the first line. The first line contains the text `# Welcome to Joplin!!`.
   - NVDA reads: "number Welcome to Joplin!!"
7. Press the up arrow.
   - This moves the cursor to the titlebar.
   - NVDA reads: "edit 1. Welcome to Joplin"
8. Press the down arrow.
   - This moves the cursor to the note body.
   - NVDA reads: "edit multi line number welcome to Joplin!!"


As mentioned in the issue description, steps 7 and 8 are potentially problematic. In particular, it isn't clear from the announced message which field has focus (title/body).

</details>
<details><summary>Windows Narrator test results</summary>

**Setup**: Joplin with a slightly modified version of the "1. Welcome to Joplin" note open in the editor. I've just disabled NVDA and enabled Windows Narrator.

**Steps**:
1. Clicked just before "Joplin" in the first paragraph.
    - This moves the cursor to the beginning of the first paragraph.
    - Narrator reads: "Joplin is a free, open source note taking and to-do application" and continues reading the note from that point.
2. Pressed the up arrow key.
    - This moves the cursor the the blank line just before the first paragraph.
    - Narrator reads: "blank"
3. Pressed the up arrow key.
    - This moves the cursor to the first line in the editor. The first line has content `# Welcome to Joplin!!`.
    - Narrator reads: "hash tag welcome to Joplin"
4. Pressed the right arrow key.
    - This moves the cursor just after the "#" in the first line.
    - Narrator reads: "space".
5. Pressed the right arrow key.
    - This moves the cursor just before the "W" in the first line.
    - Narrator reads: "cap w"
6. Pressed the control key, then the right arrow key.
    - This moves the cursor just after the word "Welcome".
    - Narrator reads: "welcome".

</details>


## Attempting to reproduce this: Rich Text Editor

So far, I've only tested the Rich Text editor with NVDA:
<details><summary>NVDA test results (Rich Text Editor)</summary>

**Setup**: I've opened a slightly modified version of the "1. Welcome to Joplin" note open in the Rich Text Edito. NVDA is enabled.

**Steps**:
1. Clicked just before the first word of the first paragraph.
   - This moves the cursor just before "Joplin" in the first paragraph.
   - NVDA reads: "Rich Text editor. Press Escape then Tab to escape focus. section  multi line  editable  Joplin is a free, open source note taking and to-do application" then continues reading the first paragraph.
2. Press the up arrow key.
   - This moves the cursor to the heading, "Welcome to Joplin!!".
   - NVDA reads: "heading level 1 Welcome to Joplin!!"
3. Press the right arrow key
   - This moves the cursor to just after the "W" in "Welcome to Joplin!!"
   - NVDA reads: "e"
4. Press the right arrow.
   - This moves the cursor to just before the "l" in "Welcome to Joplin!!"
   - NVDA reads: "l" and plays a tone.
5. Press control, then the right arrow.
   - This moves the cursor to just before "to" in "Welcome to Joplin!!"
   - NVDA reads: "to".

</details>


## Testing environment details

For comparison, I'm including additional details about my version of Joplin ([version 3.1.18](https://github.com/laurent22/joplin/releases/tag/v3.1.18)):
- **Installed plugins:** Diff view v0.1.1, Joplin debug tool v0.2.0, Backup v1.4.2. The backup plugin is disabled.
- **Markdown editor settings**:
    - In settings > general, "enable spell checking in Markdown editor" is **disabled**.
    - In settings > general > advanced, "use the legacy Markdown editor" is **disabled**.
- **Language settings**: Joplin's language is set to English (United Kingdom) and my system language is set to English (United States).

## Request for additional information

- What screen reader are you using?
- Is this on Windows 10 or Windows 11?
- Joplin's Markdown editor uses the CodeMirror library. They make a copy [of this editor available on their website](https://codemirror.net/try/). Do you experience the lack of announcements issue in a copy of this editor opened in a web browser? 
- Does restarting Joplin by closing it with File > Quit (<kbd>ctrl</kbd>+<kbd>q</kbd>), then re-launching it resolve the lack of screen reader announcements?
   - I believe I've had issues with screen reader support in Joplin in the past if I opened Joplin before starting the screen reader on Linux. If this issue has a similar cause, completely restarting Joplin may help.


Thank you again for reporting this!

**Comment by RafaelFernandesBR** [comment]
Hi, I just installed version 3.1.23 here, and the problem was fixed in this version.
The problem only appeared in the final version 22.
